src/averageAges.js

```diff
@@ -52,7 +71,18 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(child => onlyWithSon
+    ? people.find(mother => mother.name === child.mother) && child.sex === 'm'
+    : people.find(mother => mother.name === child.mother)
+  );
+
+  const ageDiff = children.map(child => {
+    return child.born - people.find(mother => {
```

Comment:
what if the result of find is undefined?

Reply:
There will be an error: TypeError: Cannot read property 'born' of undefined.
Do you mean that I should change code logic here?